recipes/soci/all/test_package/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+PROJECT(SociTest)
+cmake_minimum_required(VERSION 3.0)
+enable_testing()
+
+set(CMAKE_BUILD_TYPE "Debug")
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+
+conan_basic_setup(TARGETS)
+conan_include_build_modules()
+
+if (WITH_EMPTY)
+    add_executable(empty_exe soci.cpp)
+    target_compile_definitions(empty_exe PUBLIC BACKEND_EMPTY)
+    target_link_libraries(empty_exe PUBLIC SOCI::soci_core SOCI::soci_empty)
+    add_test(
+        NAME TestSociEmpty
+        COMMAND empty_exe
+    )
```

Comment:
Instead of testing using `add_test`, can you run the executable in the `test()` method of the test_package conanfile using `self.run(.., run_environment=True)`?
This makes this recipe look like all others.